test/feature.go

```diff
@@ -127,39 +140,22 @@ func filterFeatures(tags string, collected map[string]*gherkin.Feature) (feature
 	return features
 }
 
-func applyTagFilter(tags string, ft *gherkin.Feature) {
+func applyTagFilter(tags string, ft *feature) {
```

Comment:
Since we're going to modify the ft instance, does it make sense to scope the method to the Feature type?

```go
func (ft *Feature) applyTagFilter(tags string)
```

Also, the "tags" argument makes me thing that this is a list of tag and I think this is instead a filter (a string containing && tokens). Also, the "filter" name matches with the below method "matchesTags". What do you think about?